descwl_coadd/coadd.py

```diff
@@ -1,1236 +1,697 @@
+        weight = get_exp_weight(exp)
 
-        # we need the weights in the coadds to be the same
-        # so we replace them here
-        psf_data["weights"] = image_data["weights"]
-        noise_data["weights"] = image_data["weights"]
+        # order must match stackers, wcss, bboxes
+        exps2add = [exp, noise_exp, psf_exp]
 
-        # now we coadd
-        self.coadd_psf_exp = self._make_coadd(**psf_data)
-        pimage = self.coadd_psf_exp.image.array
-        wbad = np.where(~np.isfinite(pimage))
-        if wbad[0].size == pimage.size:
-            raise ValueError('no good pixels in the psf')
-        if wbad[0].size > 0:
-            self.log.info('zeroing %d bad psf pixels' % wbad[0].size)
-            pimage[wbad] = 0.0
+        for _stacker, _exp, _wcs, _bbox in zip(stackers, exps2add, wcss, bboxes):
+            warp_and_add(
+                _stacker, warper, _exp, _wcs, _bbox, weight,
+            )
+        nuse += 1
 
-        self.coadd_exp = self._make_coadd(**image_data)
-        self.coadd_exp.setPsf(make_stack_psf(pimage))
+    if nuse == 0:
+        return None, None, None
 
-        self.coadd_noise_exp = self._make_coadd(**noise_data)
-        self.coadd_noise_exp.setPsf(make_stack_psf(pimage))
+    stacker.fill_stacked_masked_image(coadd_exp.maskedImage)
+    noise_stacker.fill_stacked_masked_image(coadd_noise_exp.maskedImage)
+    psf_stacker.fill_stacked_masked_image(coadd_psf_exp.maskedImage)
 
-    def _make_warps(self, *, exps, dims, wcs, dopsf=False):
-        """
-        make the warp images
-        """
+    logger.info('making psf')
+    psf = extract_coadd_psf(coadd_psf_exp, logger)
+    coadd_exp.setPsf(psf)
+    coadd_noise_exp.setPsf(psf)
 
-        # Setup coadd/warp psf model
-        input_recorder_config = CoaddInputRecorderConfig()
+    return coadd_exp, coadd_noise_exp, coadd_psf_exp
 
-        input_recorder = CoaddInputRecorderTask(
-            config=input_recorder_config, name="dummy",
-        )
 
-        if dopsf:
-            coadd_psf_config = CoaddPsfConfig()
-            coadd_psf_config.warpingKernelName = self.interp
+def make_coadd_exposure(coadd_bbox, coadd_wcs):
+    """
+    make a coadd exposure with extra mask planes for
+    rejected, clipped, sensor_edge
 
-        # warp stack images to coadd wcs
-        warp_config = afw_math.Warper.ConfigClass()
+    Parameters
+    ----------
+    coadd_bbox: geom.Box2I
+        the bbox for the coadd exposure
+    coads_wcs: DM wcs
+        The wcs for the coadd exposure
 
-        # currently allows up to lanczos5, but a small change would allow
-        # higher order
-        warp_config.warpingKernelName = self.interp
-        warper = afw_math.Warper.fromConfig(warp_config)
+    Returns
+    -------
+    ExpsureF
+    """
+    coadd_exp = afw_image.ExposureF(coadd_bbox, coadd_wcs)
+    coadd_exp.mask.addMaskPlane("REJECTED")
+    coadd_exp.mask.addMaskPlane("CLIPPED")
+    coadd_exp.mask.addMaskPlane("SENSOR_EDGE")
```

Comment:
I didn't see theser flags being used anywhere? Did I miss it?

Reply:
DM adds these planes in their coadd processing, so I added them to make the default set of bits the same

Reply:
Maybe add a comment to that effect?

Reply:
done